pypeit/core/pydl.py

```diff
@@ -233,112 +233,136 @@ def __init__(self, x, nord=4, npoly=1, bkpt=None, fullbkpt = None, bkspread=1.0,
+                    bkpt = np.interp(xspot,np.arange(nx),x)
                 else:
-                    bkspace = (bkpt[1] - bkpt[0]) * np.float32(bkspread)
-                for i in np.arange(1, nord, dtype=np.float32):
-                    fullbkpt = np.insert(fullbkpt, 0, bkpt[0]-bkspace*i)
-                    fullbkpt = np.insert(fullbkpt, fullbkpt.shape[0],
-                                         bkpt[nshortbkpt-1] + bkspace*i)
-
-            #
-            # Set the attributes
-            #
-            nc = fullbkpt.size - nord
-            self.breakpoints = fullbkpt
-            self.nord = nord
-            self.npoly = npoly
-            self.mask = np.ones((fullbkpt.size,), dtype='bool')
-            if npoly > 1:
-                self.coeff = np.zeros((npoly, nc), dtype='d')
-                self.icoeff = np.zeros((npoly, nc), dtype='d')
+                    raise ValueError('No information for bkpts.')
+            imin = bkpt.argmin()
+            imax = bkpt.argmax()
+            if x.min() < bkpt[imin]:
+                if verbose:
+                    print('Lowest breakpoint does not cover lowest x value: changing.')
+                bkpt[imin] = x.min()
+            if x.max() > bkpt[imax]:
+                if verbose:
+                    print('Highest breakpoint does not cover highest x value: changing.')
+                bkpt[imax] = x.max()
+            nshortbkpt = bkpt.size
+            fullbkpt = bkpt.copy()
+            if nshortbkpt == 1:
+                bkspace = np.float32(bkspread)
             else:
-                self.coeff = np.zeros((nc,), dtype='d')
-                self.icoeff = np.zeros((nc,), dtype='d')
-            self.xmin = 0.0
-            self.xmax = 1.0
-            if 'funcname' in kwargs:
-                self.funcname = kwargs['funcname']
-            else:
-                self.funcname = 'legendre'
+                bkspace = (bkpt[1] - bkpt[0]) * np.float32(bkspread)
+            for i in np.arange(1, nord, dtype=np.float32):
+                fullbkpt = np.insert(fullbkpt, 0, bkpt[0]-bkspace*i)
+                fullbkpt = np.insert(fullbkpt, fullbkpt.shape[0],
+                                     bkpt[nshortbkpt-1] + bkspace*i)
 
-    def to_dict(self):
+        #
+        # Set the attributes
+        #
+        nc = fullbkpt.size - nord
+        self.breakpoints = fullbkpt
+        self.nord = nord
+        self.npoly = npoly
+        self.mask = np.ones((fullbkpt.size,), dtype='bool')
+        if npoly > 1:
+            self.coeff = np.zeros((npoly, nc), dtype='d')
+            self.icoeff = np.zeros((npoly, nc), dtype='d')
+        else:
+            self.coeff = np.zeros((nc,), dtype='d')
+            self.icoeff = np.zeros((nc,), dtype='d')
+        self.xmin = 0.0
+        self.xmax = 1.0
+        if 'funcname' in kwargs:
+            self.funcname = kwargs['funcname']
+        else:
+            self.funcname = 'legendre'
+
+    @classmethod
+    def from_dict(cls, from_dict):
```

Comment:
I wouldn't call the name of the function and the argument of the function the same thing.  I.e., I would change this to, e.g., `def from_dict(cls, input):`